tools/buildgen/extract_metadata_from_bazel_xml.py

```diff
@@ -567,6 +595,16 @@ def _generate_build_extra_metadata_for_tests(tests, bazel_rules):
     return test_metadata
 
 
+def _detect_and_print_issues(build_yaml_like):
+    """Try detecting some unusual situations and warn about them."""
+    for tgt in build_yaml_like['targets']:
+        if tgt['build'] == 'test':
+            for src in tgt['src']:
+                if src.startswith('src/') and not src.endswith('.proto'):
+                    print('source file from under "src/" tree used in test ' +
+                          tgt['name'] + ': ' + src)
+
+
 # extra metadata that will be used to construct build.yaml
```

Comment:
It seems strange to hard-code all of this data inside of this script instead of putting it in an external data file somewhere.  Could this be moved to build_handwritten.yaml, or something similar?

Reply:
I didn't see a response to this question.